### PR TITLE
d_a_tag_statue_evt OK

### PR DESCRIPTION
**Pull request by marwernerh**
Matching and some documentation

### CODE REVIEW

File: rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp [hunk 1]
```diff
@@ -1,805 +1,529 @@
...
 
 /* 805A6F58-805A6F78 000078 0020+00 1/1 0/0 0/0 .text            CheckCreateHeap__FP10fopAc_ac_c */
-#pragma push
-#pragma optimization_level 0
-#pragma optimizewithasm off
-static asm void CheckCreateHeap(fopAc_ac_c* param_0) {
-    nofralloc
-#include "asm/rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt/CheckCreateHeap__FP10fopAc_ac_c.s"
+static int CheckCreateHeap(fopAc_ac_c* i_this) {
+    daTagStatue_c* tag_statue = reinterpret_cast<daTagStatue_c*>(i_this);
```
Comment (hatal175) [inline rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp:11]: static_cast doesn't work?

File: rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp [hunk 2]
```diff
@@ -1,805 +1,529 @@
...
-#pragma force_active on
-SECTION_DATA static void* action_table[3] = {
-    (void*)(((char*)&d_a_tag_statue_evt__stringBase0) + 0x90),
-    (void*)(((char*)&d_a_tag_statue_evt__stringBase0) + 0x95),
-    (void*)(((char*)&d_a_tag_statue_evt__stringBase0) + 0x9D),
-};
-#pragma pop
+/* 805A7230-805A73DC 000350 01AC+00 1/1 0/0 0/0 .text            CreateHeap__13daTagStatue_cFv */
+int daTagStatue_c::CreateHeap() {
+    J3DModelData* model_data = reinterpret_cast<J3DModelData*>(dComIfG_getObjectRes(l_arcName, 6));
```
Comment (hatal175) [inline rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp:143]: static_cast should work here as well

File: rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp [hunk 3]
```diff
@@ -1,805 +1,529 @@
...
+    if(!mpModel) return 0;
 
-/* 805A8324-805A8330 000128 000C+00 0/1 0/0 0/0 .data            l_statue_ang$3944 */
-#pragma push
-#pragma force_active on
-SECTION_DATA static u8 l_statue_ang[12] = {
-    0x40, 0x00, 0x40, 0x00, 0x21, 0x25, 0x40, 0x00, 0x40, 0x00, 0x40, 0x00,
-};
-#pragma pop
+    J3DAnmTextureSRTKey* btk =  reinterpret_cast<J3DAnmTextureSRTKey*>(dComIfG_getObjectRes(l_arcName, 12));
```
Comment (hatal175) [inline rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp:148]: same

File: rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp [hunk 4]
```diff
@@ -1,805 +1,529 @@
...
+    if(!mpPlatformCircleBtk || !mpPlatformCircleBtk->init(model_data, btk, 1, 0, 1.0f, 0, -1))
+        return 0;
 
-/* 805A8380-805A838C 000184 000C+00 2/2 0/0 0/0 .data            __vt__12J3DFrameCtrl */
-SECTION_DATA extern void* __vt__12J3DFrameCtrl[3] = {
-    (void*)NULL /* RTTI */,
-    (void*)NULL,
-    (void*)__dt__12J3DFrameCtrlFv,
-};
+    J3DAnmTevRegKey* brk = reinterpret_cast<J3DAnmTevRegKey*>(dComIfG_getObjectRes(l_arcName, 9));
```
Comment (hatal175) [inline rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp:155]: same

File: rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp [hunk 5]
```diff
@@ -1,805 +1,529 @@
...
-#pragma optimization_level 0
-#pragma optimizewithasm off
-asm void daTagStatue_c::event_proc_call() {
-    nofralloc
-#include "asm/rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt/event_proc_call__13daTagStatue_cFv.s"
+/* 805A759C-805A76AC 0006BC 0110+00 1/1 0/0 0/0 .text            event_proc_call__13daTagStatue_cFv */
+void daTagStatue_c::event_proc_call() {
+    typedef void (daTagStatue_c::*actionFunc)(void);
+    static actionFunc l_func[4] = {
+    &daTagStatue_c::actionWait, &daTagStatue_c::actionOrderEvent,
```
Comment (hatal175) [inline rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp:213]: format (or at least indent

File: rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp [hunk 6]
```diff
@@ -1,805 +1,529 @@
...
 };
 
 /* 805A8258-805A8270 00005C 0018+00 0/1 0/0 0/0 .data            l_cull_box */
-#pragma push
-#pragma force_active on
-SECTION_DATA static u8 l_cull_box[24] = {
-    0xC3, 0x96, 0x00, 0x00, 0xC2, 0xC8, 0x00, 0x00, 0xC3, 0x96, 0x00, 0x00,
-    0x43, 0x96, 0x00, 0x00, 0x43, 0xC8, 0x00, 0x00, 0x43, 0x96, 0x00, 0x00,
+static Vec l_cull_box[2] = {
+    -300.0f, -100.0f, -300.0f, // Min
```
Comment (hatal175) [inline rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp:91]: Add curly braces for each Vec

File: rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp (hunk 3 above)
Comment (marwernerh) [reply rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp:148]: Unfortunately, static_cast doesn't work here; it generates regalloc issues.

File: rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp (hunk 4 above)
Comment (marwernerh) [reply rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp:155]: Unfortunately, static_cast doesn't work here; it generates regalloc issues.

**Review by hatal175 — APPROVED** [review]

File: rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp [hunk 7]
```diff
@@ -1,805 +1,527 @@
...
+            found_owl_statue_xy_pos.y = 0.0f;
+            found_owl_statue_xy_pos.z = found_owl_statue_z;
+
+            const f32 statue_tag_z = fopAcM_GetPosition(this).z;
+            const f32 statue_tag_x = fopAcM_GetPosition(this).x;
+            statue_tag_xy_pos.x = statue_tag_x;
+            statue_tag_xy_pos.y = 0.0f;
+            statue_tag_xy_pos.z = statue_tag_z;
+
+            const f32 distance_between_statue_and_tag = sqrtf(PSVECSquareDistance(&found_owl_statue_xy_pos, &statue_tag_xy_pos));
```
Comment (GinNoOokami) [inline rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp:426]: Try using cXyz::abs() here instead.

**Review by GinNoOokami — COMMENTED** [review]
Don't forget to remove the asms with `./tp remove-unused-asm` when completing a file :)

**Comment by hatal175** [comment]
Merged already. Oh well. Next person gets bonus files :)

File: rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp (hunk 7 above)
Comment (GinNoOokami) [reply rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp:426]: There's another instance of this in the next function as well. The abs() call should inline to the same thing here, and is likely what was originally used, so give it a try and see if it still matches.

File: rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp (hunk 3 above)
Comment (GinNoOokami) [reply rel/d/a/tag/d_a_tag_statue_evt/d_a_tag_statue_evt.cpp:148]: `static_cast` does, but it looks like C-style casts work here. I think that might be preferable to a `reinterpret_cast`, but maybe that's just me.

**Pull request merged**